spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -50,14 +69,29 @@ object EmbeddedCassandra {
 
 }
 
-private[connector] class CassandraRunner(val configTemplate: String) extends Embedded {
+object SecondEmbeddedCassandra {
+
+  val DefaultHost = "127.0.0.2"
+  final val DefaultNativePort = 9043
+
+  val HostProperty = "IT_TEST_CASSANDRA_HOST2"
+
+  private[connector] var cassandra: Option[CassandraRunner] = None
+
+  val cassandraHost = InetAddress.getByName(sys.env.getOrElse(HostProperty, DefaultHost))
+
+  Runtime.getRuntime.addShutdownHook(new Thread(new Runnable {
+    override def run() = cassandra.map(_.destroy())
+  }))
+
+}
+
+private[connector] class CassandraRunner(val configTemplate: String, host: String, nativePort: Integer) extends Embedded {
 
   import java.io.{File, FileOutputStream, IOException}
   import org.apache.cassandra.io.util.FileUtils
   import com.google.common.io.Files
-  import EmbeddedCassandra._
 
-  final val DefaultNativePort = 9042
   val tempDir = mkdir(new File(Files.createTempDir(), "cassandra-driver-spark"))
```

Comment:
Have you seen dir/file collisions when running both against this same class? wouldn't both be using the same commit log and data dirs? I didn't see those changed in their respective yamls